source/common/grpc/common.cc

```diff
@@ -160,12 +160,18 @@ std::chrono::milliseconds Common::getGrpcTimeout(const Http::RequestHeaderMap& r
   std::chrono::milliseconds timeout(0);
   const Http::HeaderEntry* header_grpc_timeout_entry = request_headers.GrpcTimeout();
   if (header_grpc_timeout_entry) {
-    uint64_t grpc_timeout;
-    // TODO(dnoe): Migrate to pure string_view (#6580)
-    std::string grpc_timeout_string(header_grpc_timeout_entry->value().getStringView());
-    const char* unit = StringUtil::strtoull(grpc_timeout_string.c_str(), grpc_timeout);
-    if (unit != nullptr && *unit != '\0') {
-      switch (*unit) {
+    int64_t grpc_timeout;
+    absl::string_view timeout_entry = header_grpc_timeout_entry->value().getStringView();
```

Comment:
random thought: `absl::SimpleAtoi` below allows/ignores whitespace, did someone check the definition of whitespace matches the one that the trimming in the header parsing takes care of? It might be defensive to trim ourselves here, otherwise some of the assumptions about first/last characters might leave wiggle room for nasty edge cases.